xarray/core/dataarray.py

```diff
@@ -215,27 +215,110 @@ def __setitem__(self, key, value) -> None:
+      dictionary: ``x.attrs``
     - Convert to a pandas Series: ``x.to_series()``.
 
-    Getting items from or doing mathematical operations with a DataArray
-    always returns another DataArray.
+    Getting items from or doing mathematical operations with a
+    DataArray always returns another DataArray.
+
+    Parameters
+    ----------
+    data : array_like
+        Values for this array. Must be an ``numpy.ndarray``, ndarray
+        like, or castable to an ``ndarray``. If a self-described xarray
+        or pandas object, attempts are made to use this array's
+        metadata to fill in other unspecified arguments. A view of the
+        array's data is used instead of a copy if possible.
+    coords : sequence or dict of array_like, optional
+        Coordinates (tick labels) to use for indexing along each
+        dimension. The following notations are accepted:
+
+        - mapping {dimension name: array-like}
+        - sequence of tuples that are valid arguments for
+          ``xarray.Variable()``
+          - (dims, data)
+          - (dims, data, attrs)
+          - (dims, data, attrs, encoding)
+
+        Additionally, it is possible to define a coord whose name
+        does not match the dimension name, or a coord based on multiple
+        dimensions, with one of the following notations:
+
+        - mapping {coord name: DataArray}
+        - mapping {coord name: Variable}
+        - mapping {coord name: (dimension name, array-like)}
+        - mapping {coord name: (tuple of dimension names, array-like)}
+
+    dims : hashable or sequence of hashable, optional
+        Name(s) of the data dimension(s). Must be either a hashable
+        (only for 1D data) or a sequence of hashables with length equal
+        to the number of dimensions. If this argument is omitted,
+        dimension names default to ``['dim_0', ... 'dim_n']``.
+    name : str or None, optional
+        Name of this array.
+    attrs : dict_like or None, optional
+        Attributes to assign to the new instance. By default, an empty
+        attribute dictionary is initialized.
+
+    Examples
+    --------
+    Import modules:
+
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> import xarray as xr
+
+    Create data:
+
+    >>> np.random.seed(0)
+    >>> temperature = 15 + 8 * np.random.randn(2, 2, 3)
+    >>> precipitation = 10 * np.random.rand(2, 2, 3)
+    >>> lon = [[-99.83, -99.32], [-99.79, -99.23]]
+    >>> lat = [[42.25, 42.21], [42.63, 42.59]]
+    >>> time = pd.date_range("2014-09-06", periods=3)
+    >>> reference_time = pd.Timestamp("2014-09-05")
+
+    Initialize a dataarray with multiple dimensions:
+
+    >>> da = xr.DataArray(
+    ...     data = temperature,
+    ...     dims = ["x", "y", "time"],
+    ...     coords={
+    ...         "lon": (["x", "y"], lon),
+    ...         "lat": (["x", "y"], lat),
+    ...         "time": time,
+    ...         "reference_time": reference_time,
+    ...     },
+    ... )
```

Comment:
Can we show `da` here?

Reply:
Not sure I follow, what should have a da?

Reply:
Could we have: `>>> da` to print what the array looks like?

Reply:
Done, it makes sense to show how the original dataset is supposed to look like. 

Though I'm a little reluctant of doing it, because the dataset print has line lengths that is a little longer than what pep8 recommends.

Reply:
That's fine — we actually disable the PEP8 line checks in preference of the black ones (your editor / IDE should be picking these configs up — lmk if you get warnings that you'd like to disable).

If it causes any checks to fail, we can skip the checks for this section, it's a good tradeoff!